### PR TITLE
use the haveibeenpwned.com API to test new passwords

### PR DESCRIPTION
**Pull request by Schparky**
Found a HIBP PHP library that is a better fit. Uses Guzzle and implements the API more fully. Doesn't lock to a specific API version, though.

### CODE REVIEW

File: application/common/models/Password.php [hunk 1]
```diff
@@ -241,4 +247,28 @@ public function validateNotPublicPassword($attribute)
             }
         }
     }
+
+    /**
+     * @param string $attribute
+     */
+    public function validateNotBeenPwned($attribute)
```
Comment (wcjr) [inline application/common/models/Password.php]: what is `$attribute` here?  Is it a password?

Comment (Schparky) [reply application/common/models/Password.php]: This is a Yii2 validator, so the required parameter is the name of the attribute being validated. In this case, `$attribute == 'password'`

**Comment by Schparky** [comment]
> I'd suggest creating an issue in their repo for versioning the api if possible.

Unfortunately, the source is in a repository on a privately-hosted site that uses Gitlab Community Edition. I sent an email, but haven't heard anything back. Depending on how important this is, I could fork it. I did make sure to catch Guzzle exceptions and not re-throw them, so even if the API changes the end user experience should not break.

File: application/common/models/Password.php (hunk 1 above)
Comment (wcjr) [reply application/common/models/Password.php]: gotcha, that does ring a bell, thanks.

**Comment by wcjr** [comment]
I can't predict how important at this point a versioned api will be, this will come down to a judgment call of how much we trust this author.  Your call man, you have my support either way.

**Comment by forevermatt** [comment]
This looks good to me. I didn't realize it ahead of time, but sending a password (even lightly hashed) to someone else's library makes me slightly nervous. However, though his repo has very few downloads / stars, looking through the PwnedPasswords sections of his code it all seems fine at the moment.

@Schparky I'm a little confused by your comment about the code being in a privately hosted site... it looks like [the Packagist listing for it](https://packagist.org/packages/icawebdesign/hibp-php) points to a simple [GitHub repo](https://github.com/icawebdesign/hibp-php).

And, after looking some more at his code (and the API docs), it looks like I was mistaken about the API versioning issue. HIBP's main API is versioned, but the PwnedPasswords API doesn't seem to be versioned, so it's something of a non-issue apparently.

**Comment by Schparky** [comment]
> @Schparky I'm a little confused by your comment about the code being in a privately hosted site... it looks like [the Packagist listing for it](https://packagist.org/packages/icawebdesign/hibp-php) points to a simple [GitHub repo](https://github.com/icawebdesign/hibp-php).

Wierd, I can only assume that he moved it since yesterday. I did send an email asking if he had it on a public site as well.

**Pull request merged**